### DISCUSSION TITLE
questions

### DISCUSSION THREAD

**Discussion by hemna** [LeeOtts/Ender3v2-Klipper-Configs#6] [Q&A] [answered]
I'm trying to use your setup for my ender 3v2 v4.2.2 board with the sprite pro extruder and the CRtouch.

In the extruder section of your printer.cfg, why do you have this commented out?

```
#control: pid
#pid_Kp: 23.561
#pid_Ki: 1.208
#pid_Kd: 114.859
```

I can't get klipper to start with that commented out. I just uncommented it to get klipper happy.

Also, you have the following commented out as well in the bltouch section ?

```
[bltouch]
sensor_pin: ^PB1
control_pin: PB0
x_offset: -31.8
y_offset: -40.5
#z_offset: 4
```

**Comment by LeeOtts** [comment, marked as answer]
The pid tuning gets commented out when you do a pid tuning and placed in the Save Config at the very bottom. 

Same with the commented-out z offset. It is put in the Save Config at the bottom when you calibrate a z offset. 

Did you copy the very bottom of the printer config that says Save Config as well?

**Comment by LeeOtts** [comment]
Hello Hemma,

I apologize for the short response earlier. I was on my lunch break when I replied, and I understand how frustrating it can be to receive a brief answer.

I have converted this issue into a discussion since it appears that you are not facing any issues with Klipper if you're using the complete printer.cfg. I'd be happy to provide you with further clarification on this.

Towards the bottom of the printer.cfg, you'll find a section named.
 "#*# <---------------------- SAVE_CONFIG ---------------------->".
 I will attach a image at the bottom to show the section I am talking about

 This is an autogenerated section by Klipper that contains the configurations in question.  

Klipper automatically commented out those sections when I made adjustments to my z-offset and pid-tuned my hotend.

You are correct in saying that these configurations are essential for Klipper to function correctly, and they were present when I first booted Klipper. However, it's possible that you might have accidentally left out the SAVE_CONFIG section while trying to run it on a different board.

It's worth noting that every time you update your z-offset or perform PID tuning, the numbers get updated in the SAVE_CONFIG section. .

Once you've got it up and running, run a PID, and the information will automatically be added to the SAVE_CONFIG section.

I hope this explanation helps you resolve the issue. 

Please let me know if you have any further questions or concerns.

![save_config](https://github.com/LeeOtts/Ender3v2-Klipper-Configs/assets/91760497/24a94e83-ae1d-47df-8b98-9b996a707fc2)

**Comment by master-of-the-dungeon** [comment]
Trying to use your macros.cfg, but every time i get Move exceeds maximum extrusion (1.011mm^2 vs 0.640mm^2). I use CURA. What should i do? Thank you!

**Reply by LeeOtts** [reply]
You can add this to your extruder section in the printer config. 

max_extrude_cross_section: 5
